### PR TITLE
feat: Implemented rabbitmq support

### PR DESCRIPTION
**Pull request by hofmeister**

### CODE REVIEW

File: src/main/java/com/kapeta/spring/annotation/KapetaEnableRabbitMQ.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+package com.kapeta.spring.annotation;
+
+import com.kapeta.spring.rabbitmq.RabbitBeanConfig;
+import com.kapeta.spring.rabbitmq.RabbitHealthConfiguration;
+import org.springframework.context.annotation.Import;
+
+import java.lang.annotation.*;
+
+/**
+ * Add this to your Application class to enable mongodb support
```
Comment (sorenmat) [inline src/main/java/com/kapeta/spring/annotation/KapetaEnableRabbitMQ.java]: I guess rabbit :)

File: src/main/java/com/kapeta/spring/rabbitmq/KapetaRabbitConnectionFactory.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+import java.util.UUID;
+
+import static com.kapeta.spring.rabbitmq.RabbitHelper.defaultRetryTemplate;
+
+@Slf4j
+public class KapetaRabbitConnectionFactory {
+
+    public static final String PORT_AMQP = "amqp";
+
+    private static final int MAX_ATTEMPTS = 100;
```
Comment (sorenmat) [inline src/main/java/com/kapeta/spring/rabbitmq/KapetaRabbitConnectionFactory.java:34]: Should this be configurable?

File: src/main/java/com/kapeta/spring/rabbitmq/RabbitBeanConfig.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            RabbitConfigurationManager config,
+            RabbitManagementService managementService,
+            KapetaRabbitConnectionFactory connectionFactory) {
+        return new RabbitConnectionManager(config, managementService, connectionFactory);
+    }
+
+
+    @Bean
+    private MessageConverter messageConverter(ObjectMapper objectMapper) {
+        return new Jackson2JsonMessageConverter(objectMapper, "com.kapeta.*");
```
Comment (sorenmat) [inline src/main/java/com/kapeta/spring/rabbitmq/RabbitBeanConfig.java]: Is this too limiting?

Comment (hofmeister) [reply src/main/java/com/kapeta/spring/rabbitmq/RabbitBeanConfig.java]: Actually need to get rid of this - unused

File: src/main/java/com/kapeta/spring/rabbitmq/KapetaRabbitConnectionFactory.java (hunk 2 above)
Comment (hofmeister) [reply src/main/java/com/kapeta/spring/rabbitmq/KapetaRabbitConnectionFactory.java:34]: It's only used for an internal ensure vhost thing so figured it'd be something people wont change

**Review by sorenmat — APPROVED** [review]
LGTM

**Pull request merged**